idptoken/introspector.go

```diff
@@ -476,6 +498,30 @@ func (i *Introspector) getURLForIssuerWithCallback(ctx context.Context, issuer s
 	return i.trustedIssuerNotFoundFallback(ctx, i, issuer)
 }
 
+// applyScopeFilterIfNeeded filters the scope of the introspection result
+// and preserves policies only that match the filter if it is set.
+// It's used just in case when the scope filtering is not done on the introspection endpoint side.
+func (i *Introspector) applyScopeFilterIfNeeded(res *IntrospectionResult) {
```

Comment:
It seems this function would be better placed in the IntrospectionResult type with the signature ApplyScopeFilter(scopeFilter). Additionally, I noticed identical code in the API Gateway. By moving this function into the IntrospectionResult type, it would be reusable for other clients.

Reply:
Good point! Reworked